src/config_test.ts

```diff
@@ -322,6 +326,7 @@ describe('KubeConfig', () => {
             };
 
             assertRequestOptionsEqual(opts, expectedOptions);
+            strictEqual((requestContext.getAgent()! as any).options.servername, 'kube.example2.com');
```

Comment:
Isn't this rather an `https.Agent` than an `any?`
This would remove the unnecessary type casting but you probably still need the bang-operator or a different check.
Not sure if it's worth it, just wanted to mention that.

Reply:
options isn't exposed as public (at least not in the typings), so Typescript doesn't like it. The cast as `any` is mostly to get the typescript compiler to leave it alone.